src/Components/Components/src/RenderModeAttribute.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Specifies a fixed rendering mode for a component type.
+///
+/// Where possible, components should not specify any render mode this way, and should
+/// be implemented to work across all render modes. Component authors should only specify
+/// a fixed rendering mode when the component is incapable of running in other modes.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class)]
+public abstract class RenderModeAttribute : Attribute
```

Comment:
Do we need to allow multiple for this?

Reply:
I don't think so. What would we do if there were multiple?

Remember this is about specifying a *required* rendermode, not a set of allowed rendermodes (which, if implemented in the future, would be a separate API).

Reply:
I mean AllowMultiple = false

Reply:
That's the default. If I'm missing something and it has to be specified explicitly in this case, please let me know!